resources/js/common.js

```diff
@@ -283,3 +282,40 @@ function postwith (to,p) {
   document.body.removeChild(myForm) ;
 }
 
+
+function isValidDate(dateString)

```

Comment:
Yes, it's really handy and better to use JQuery/Plugin, however at the time when this PR was being written, the version of jQuery  was found really behind.    Now, after pulling all latest commits into this PR, I decide to completely give up date validation because Subscription dates still look not required and can underlying mySQL can blindly set all invalid date inputs to   01/01/1970 after they are are submitted and can be changed at any time to good ones.